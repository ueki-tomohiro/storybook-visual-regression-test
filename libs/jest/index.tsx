import { composeStories, StoryFile } from "@storybook/testing-react";
import { render } from "@testing-library/react";
import React from "react";

export const renderFromStories = (stories: StoryFile) => {
  const composed = composeStories(stories);
  const testCases = Object.values(composed).map((Story: any) => [Story.storyName, Story, Story.args]);
  test.each(testCases)("renders %s", (_storyName, Component: any, args) => {
    if (_storyName.match(/Fake/)) return;

    const tree = render(<Component {...(args || {})} />);
    expect(tree.baseElement).toMatchSnapshot();
  });
};
