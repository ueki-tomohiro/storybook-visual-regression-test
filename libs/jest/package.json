{
  "name": "@demo-libs/jest",
  "version": "0.0.1",
  "private": true,
  "description": "Demo Jest",
  "license": "MIT",
  "devDependencies": {
    "@babel/core": "^7.21.4",
    "@babel/preset-env": "^7.21.4",
    "@babel/preset-typescript": "^7.21.4",
    "@testing-library/react": "^14.0.0",
    "@testing-library/user-event": "^14.5.1",
    "@types/jest": "^29.5.1",
    "babel-jest": "^29.5.0",
    "babel-plugin-transform-import-meta": "^2.2.0",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^29.5.0",
    "jest-environment-jsdom": "^29.5.0"
  },
  "engines": {
    "node": "^20",
    "yarn": ">=1"
  },
  "volta": {
    "node": "20.0.0",
    "yarn": "1.22.19"
  }
}
