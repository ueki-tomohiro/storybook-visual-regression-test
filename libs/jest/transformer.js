const babelJest = require("babel-jest");
const babelOptions = {
  presets: [
    [
      "@babel/env",
      {
        targets: {
          node: "current",
        },
        exclude: ["transform-regenerator"],
      },
    ],
    "@babel/preset-typescript",
    "@babel/react",
  ],
  plugins: ["babel-plugin-transform-import-meta"],
  babelrc: false,
};
module.exports = babelJest.createTransformer(babelOptions);
