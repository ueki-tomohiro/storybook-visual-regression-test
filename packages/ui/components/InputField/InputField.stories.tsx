import { action } from "@storybook/addon-actions";
import React from "react";

import { InputField, InputFieldProps } from "./InputField";

const defaultArgs: InputFieldProps = {
  children: <></>,
  onClick: action("onClick"),
};

export default {
  title: "components/InputField",
  component: InputField,
};

export const Default = { args: defaultArgs };
