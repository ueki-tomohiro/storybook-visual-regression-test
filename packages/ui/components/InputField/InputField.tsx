import React, { ComponentPropsWithoutRef, forwardRef } from "react";

export type InputFieldProps = ComponentPropsWithoutRef<"input"> & {
  label?: string;
  error?: string;
};

export const InputField = forwardRef<HTMLInputElement, InputFieldProps>(
  ({ label, error, type = "text", value, children, ...props }, ref) => {
    return (
      <>
        <div className="relative w-full">
          <fieldset
            className={`w-full ${label ? "h-12" : "mt-1 h-11"} mb-6 rounded border border-solid ${
              error ? "border-caution" : "border-tertiary"
            } focus-within:border-secondary ${
              error ? "[&_label]:[&:not(:focus-within)]:text-caution" : "[&_label]:[&:not(:focus-within)]:text-tertiary"
            } [&_label]:focus-within:text-secondary`}
          >
            {label && (
              <legend className="min-h-3 ml-2 flex px-0.5">
                <label className={`flex text-xs`}>{label}</label>
              </legend>
            )}
            <div className={`absolute inset-0 flex items-center${label && "pt-1.5"}`}>
              <input
                {...props}
                className="bg-reset text-tertiary focus:text-secondary h-11 w-full px-2 focus:outline-none"
                type={type}
                defaultValue={value}
                ref={ref}
              />
            </div>
            <div className="-z-1 absolute inset-0" />
          </fieldset>
        </div>
        {error && <div className="text-caution ml-2 mt-2 text-sm">{error}</div>}
      </>
    );
  }
);
InputField.displayName = "InputField";
