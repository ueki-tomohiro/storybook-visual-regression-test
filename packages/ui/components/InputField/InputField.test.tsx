import { renderFromStories } from "@demo-libs/jest";

import * as stories from "./InputField.stories";

describe(stories.default.title, () => {
  renderFromStories(stories);
});
