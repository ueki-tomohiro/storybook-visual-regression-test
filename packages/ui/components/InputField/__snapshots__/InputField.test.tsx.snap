// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/InputField renders Default 1`] = `
<body>
  <div>
    <div
      class="relative w-full"
    >
      <fieldset
        class="w-full mt-1 h-11 mb-6 rounded border border-solid border-tertiary focus-within:border-secondary [&_label]:[&:not(:focus-within)]:text-tertiary [&_label]:focus-within:text-secondary"
      >
        <div
          class="absolute inset-0 flex items-centerundefined"
        >
          <input
            class="bg-reset text-tertiary focus:text-secondary h-11 w-full px-2 focus:outline-none"
            type="text"
            value=""
          />
        </div>
        <div
          class="-z-1 absolute inset-0"
        />
      </fieldset>
    </div>
  </div>
</body>
`;
